src/os/layer/preset/layerpresetmanager.js

```diff
@@ -75,14 +100,186 @@ class LayerPresetManager extends Disposable {
+   * @protected
+   */
+  onLayerStyleChanged(layerId, layer, check, event) {
+    let [,, meta] = /** @type {!Array<*>} */ (this.presets_.entry(layerId));
+    meta = /** @type {!LayerPresetsMetaData} */ (meta);
+
+    // get the debouncer for this layerId
+    if (meta && !meta.debouncer.isDisposed()) {
+      meta.debouncer.fire(check, event);
+    }
+  }
+
+  /**
+   * Handle layer style changed.
+   * @param {!string} layerId The layer id.
+   * @param {!OlLayer} layer The layer.
+   * @param {!boolean} check True if the current style needs to be tested extensively against the Preset style
+   * @param {!PropertyChangeEvent} event The layer event.
+   * @private
+   */
+  onLayerStyleChanged_(layerId, layer, check, event) {
+    // check that the style does/doesn't match the Preset. If not, set the clean over to false and stop listening
+    if (layer && event['getProperty'] && event.getProperty() == 'style') {
+      if (this.isLayerStyleDirty_(layerId, layer, check, event)) {
+        let [,, meta] = /** @type {!Array<*>} */ (this.presets_.entry(layerId));
+        meta = /** @type {!LayerPresetsMetaData} */ (meta);
+
+        const key = /** @type {OlEventsKey} */ (meta.listener);
+        if (key && layer) {
+          olEvents.unlistenByKey(key);
+          meta.listener = null;
         }
+
+        OsLayerPreset.setSavedPresetClean(layerId, false);
+        OsStyle.notifyStyleChange(layer);
+      }
+    }
+  }
+
+  /**
+   * Handle layer style changed.
+   * @param {!string} layerId The layer id.
+   * @param {!OlLayer} layer The layer.
+   * @param {!boolean} check True if the current style needs to be tested extensively against the Preset style
+   * @param {!PropertyChangeEvent} event The layer event.
+   * @return {boolean} true if the style is what the preset says it should be
+   * @private
+   */
+  isLayerStyleDirty_(layerId, layer, check, event) {
+    let dirty = false;
+    let [,, meta] = /** @type {!Array<*>} */ (this.presets_.entry(layerId));
+    meta = /** @type {!LayerPresetsMetaData} */ (meta);
+
+    if (meta && meta.selected) {
+      // check the layer configs
+      const config1 = meta.selected.layerConfig;
+      const config2 = /** @type {ILayer} */ (layer).persist();
+
+      // "basic" preset doesn't have all the details of a real layer
+      if (meta.selected.id == OsLayerPreset.DEFAULT_PRESET_ID) {
+        config2['icon'] = null;
+        delete config2['colorModel'];
+        delete config2['replaceStyle'];
+        delete config2['showRotation'];
+      }
+
+      // TODO remove this after fixing the "layer.restore()" to treat -no value- as a "clear"
+      // if not applying a new "border" style or "unique identifier", ignore the current one
+      if (!config1['lineDash']) {
+        delete config2['lineDash'];
+      }
+      if (!config1['uniqueId']) {
+        delete config2['uniqueId'];
+      }
+      if (!config1['threshold'] && config2['threshold']) {
```

Comment:
These keys feel too specific for the preset manager, and some aren't from OpenSphere. We may need a way for plugins to contribute these updates.

Reply:
I _could_ do a key-by-key comparison (which is more or less what I'm doing by deleting things that I don't want to compare at the moment). It'd be a lot slower, but also more generic. I'll think about that a bit more...